sdk/src/main/java/io/radar/sdk/Radar.kt

```diff
@@ -891,6 +909,128 @@ object Radar {
         )
     }
 
+    /**
+     * Gets the device's current location, then searches for points near that location, sorted by distance.
+     *
+     * @param[radius] The radius to search, in meters. A number between 100 and 10000.
+     * @param[tags] An array of tags to filter. See [](TODO: doc link)
+     * @param[limit] The max number of places to return. A number between 1 and 100.
+     * @param[callback] A callback.
+     */
+    @JvmStatic
+    fun searchPoints(
+        radius: Int,
+        tags: Array<String>?,
+        limit: Int?,
+        callback: RadarSearchPointsCallback
+    ) {
+        if (!initialized) {
+            callback.onComplete(RadarStatus.ERROR_PUBLISHABLE_KEY)
+
+            return
+        }
+
+        locationManager.getLocation(object : RadarLocationCallback {
+            override fun onComplete(status: RadarStatus, location: Location?, stopped: Boolean) {
+                if (status != RadarStatus.SUCCESS || location == null) {
+                    callback.onComplete(status)
+
+                    return
+                }
+
+                apiClient.searchPoints(location, radius, tags, limit, object : RadarApiClient.RadarSearchPointsApiCallback {
+                    override fun onComplete(status: RadarStatus, res: JSONObject?, points: Array<RadarPoint>?) {
+                        callback.onComplete(status, location, points)
+                    }
+                })
+            }
+        })
+    }
+
+    /**
+     * Gets the device's current location, then searches for points near that location, sorted by distance.
+     *
+     * @param[radius] The radius to search, in meters. A number between 100 and 10000.
+     * @param[tags] An array of tags to filter. See [](https://radar.io/documentation/points)
+     * @param[limit] The max number of places to return. A number between 1 and 100.
+     * @param[block] A block callback.
+     */
+    fun searchPoints(
+        radius: Int,
+        tags: Array<String>?,
+        limit: Int?,
+        block: (status: RadarStatus, location: Location?, points: Array<RadarPoint>?) -> Unit
+    ) {
+        searchPoints(
+            radius,
+            tags,
+            limit,
+            object : RadarSearchPointsCallback {
+                override fun onComplete(status: RadarStatus, location: Location?, points: Array<RadarPoint>?) {
+                    block(status, location, points)
+                }
+            }
+        )
+    }
```

Comment:
Why we need this wrapper?

Reply:
good question, I've always been confused by needing to have two signatures for each API we introduce 🤔

Reply:
@pingxia0823 @jsani-radar This exposes a block callback in Kotlin. Looks like
```
Radar.searchPoints(radius, tags, limits) { status, location, points ->
  // ...
```
instead of
```
Radar.searchPoints(radius, tags, limits, new RadarCallback() {
  @Override
  public void onComplete(RadarStatus status, Location location, RadarPoint[] points) {
    // ...
  }
});
```

Reply:
interesting, I guess it seems a bit tedious to write another signature to achieve the former - would be nice if the former were able to coalesce to the latter automagically